rspace/src/test/scala/coop/rchain/rspace/history/TrieStructureTests.scala

```diff
@@ -0,0 +1,194 @@
+package coop.rchain.rspace.history
+
+import java.nio.ByteBuffer
+
+import org.lmdbjava.Txn
+import scodec.Codec
+import scodec.bits.ByteVector
+import scodec.codecs._
+
+class TrieStructureTests
+    extends HistoryTestsBase[Txn[ByteBuffer], TestKey, ByteVector]
+    with WithLMDBStore {
+
+  implicit val codecByteVector: Codec[ByteVector] = variableSizeBytesLong(int64, bytes)
+
+  def withTrie[R](f: Trie[TestKey, ByteVector] => R): R =
+    withTestTrieStore { store =>
+      store.withTxn(store.createTxnRead()) { txn =>
+        val trieOpt = store.get(txn, store.workingRootHash.get)
+        trieOpt should not be empty
+        f(trieOpt.get)
+      }
+    }
+
+  def withTrieTxnAndStore[R](
+      f: (ITrieStore[Txn[ByteBuffer], TestKey, ByteVector],
+          Txn[ByteBuffer],
+          Trie[TestKey, ByteVector]) => R): R =
+    withTestTrieStore { store =>
+      store.withTxn(store.createTxnRead()) { txn =>
+        val trieOpt = store.get(txn, store.workingRootHash.get)
+        trieOpt should not be empty
+        f(store, txn, trieOpt.get)
+      }
+    }
+
+  "insert's effect" should "be visible in the outer read transaction" ignore {
+    withTrieTxnAndStore { (store, txn, trie) =>
+      val f = singleElementFixture
+      import f._
+
+      insert(store, key1, val1)
+      // Insert was made in a nested transaction, so it's effect should be visible
+      store.get(txn, store.workingRootHash.get) should not be None
+    }
+  }
+
+  behavior of "A trie"
+  it should "be created as an empty pointer block" in
+    withTrie {
+      case Node(PointerBlock(vector)) =>
+        vector should have size 256
+        vector should contain only None
+      case _ => fail("expected a node")
+    }
+
+  private[this] def singleElementFixture = new {
```

Comment:
How about making this a `val`?  

Then you could just `import singleElementFixture._` below...

Also, I thought "fixtures" were things like:
```scala
def withBlah[R](blah: Blah)(f: Blah => R): R
```

Is there a more general sensibility of the word I haven't picked up on?

In my mind this could just be `object SingleElementTestData`...

Reply:
Yeah, I can make it a `val` - a `def` would make sense if there weren't only constants inside the `new` block.

Regarding naming: I based my approach on the scalatest docs. Check out the first example.
http://www.scalatest.org/user_guide/sharing_fixtures

Reply:
Changed